platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/collections/ReproducibleDirectoryWalker.java

```diff
@@ -85,5 +84,35 @@ private static FileVisitResult visit(Path path, PathVisitor pathVisitor) throws
         }
     }
 
-    private final static Comparator<Path> FILES_FIRST = Comparator.<Path, Boolean>comparing(x -> x.toFile().isDirectory()).thenComparing(Path::toString);
+    private static PathWithAttributes toPathWithAttributes(Path path) {
+        try {
+            BasicFileAttributes attributes = Files.readAttributes(path, BasicFileAttributes.class);
+            return new PathWithAttributes(path, attributes, null);
+        } catch (IOException ignored) {
```

Comment:
:x: I understand this is code moved around, but I think it would be good to explain why/how this exception is ignored.

Reply:
It looks like we use similar logic as in OpenJDK:
https://github.com/openjdk/jdk/blob/1501a5e41e59162a374cf5b8cfc37faced48a6ed/src/java.base/share/classes/java/nio/file/FileTreeWalker.java#L212-L225

Added a comment